File: Irrgarten/src/irrgarten/Shield.java
package irrgarten;

/**
 * Represents a shield in the game Irrgarten
 */
public class Shield extends CombatElement {
    /**
     * Parameterized constructor to create a Shield object.
     * 
     * @param protection the protection that the shield offers
     * @param uses the number of uses left for the shield
     */
    Shield(float protection, int uses){
        super(protection,uses);
    }
    
    /**
     * Returns the protection that the shield offers. If the shield has no uses left, the protection is 0.
     * 
     * @return the protection value as a float
     */
    public float protect(){
        return this.produceEffect();
    }
    
    /**
     * Returns a string representation of the shield.
     * 
     * @return a string with information about the shield in the format "S[protection,uses]"
     */
    @Override
    public String toString(){
        return "S"+super.toString();
    }
}

File: Irrgarten_ruby/shield.rb
#encoding:utf-8
require_relative 'dice'
require_relative 'combat_element'
module Irrgarten
    # This class represents a shield from the game Irrgarten
    class Shield < CombatElement

        # Return the shield protection. If there are no uses left, it is 0
        #
        # @return [Float] The shield protection
        def protect
            self.produce_effect
        end

        # Makes a string that contains information about the object
        #
        # @return [String] A string with the info of the shield
        def to_s
            "S#{super}"
        end
    end
end
